R/get.R

```diff
@@ -262,5 +262,155 @@ get_samples.estimate_infections <- function(object, ...) {
+  # Only add if not already present
+  if (!"date" %in% names(samples)) samples[, date := as.Date(NA)]
+  if (!"strat" %in% names(samples)) samples[, strat := NA_character_]
+  if (!"time" %in% names(samples)) samples[, time := NA_integer_]
+  if (!"type" %in% names(samples)) samples[, type := NA_character_]
+
+  # Reorder columns
+  data.table::setcolorder(
+    samples,
+    c("date", "variable", "strat", "sample", "time", "value", "type")
+  )
+
+  samples[]
+}
+
+#' @rdname get_samples
+#' @export
+get_samples.forecast_secondary <- function(object, ...) {
+  data.table::copy(object$samples)
+}
+
+#' Get predictions from a fitted model
+#'
+#' @description `r lifecycle::badge("stable")`
+#' Extracts predictions from a fitted model, combining observations with model
+#' estimates. For `estimate_infections()` returns predicted reported cases, for
+#' `estimate_secondary()` returns predicted secondary observations.
+#'
+#' @param object A fitted model object (e.g., from `estimate_infections()` or
+#'   `estimate_secondary()`)
+#' @param CrIs Numeric vector of credible intervals to return. Defaults to
+#'   c(0.2, 0.5, 0.9).
+#' @param ... Additional arguments (currently unused)
+#'
+#' @return A `data.table` with columns including date, observations, and summary
+#'   statistics (mean, sd, credible intervals) for the model predictions.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' # After fitting a model
+#' predictions <- get_predictions(fit)
+#' }
+get_predictions <- function(object, ...) {
+  UseMethod("get_predictions")
+}
+
+#' @rdname get_predictions
+#' @export
+get_predictions.estimate_infections <- function(object,
+                                                CrIs = c(0.2, 0.5, 0.9),
+                                                ...) {
+  # Get samples for reported cases
+  samples <- get_samples(object)
+  reported_samples <- samples[variable == "reported_cases"]
+
+  # Calculate summary measures
+  predictions <- calc_summary_measures(
+    reported_samples,
+    summarise_by = "date",
+    order_by = "date",
+    CrIs = CrIs
+  )
+
+  # Merge with observations
+  predictions <- data.table::merge.data.table(
+    object$observations[, .(date, confirm)],
+    predictions,
+    by = "date",
+    all = TRUE
+  )
+
+  predictions
+}
+
+#' @rdname get_predictions
+#' @export
+get_predictions.estimate_secondary <- function(object,
+                                               CrIs = c(0.2, 0.5, 0.9),
+                                               ...) {
+  # Get samples for simulated secondary observations
+  samples <- get_samples(object)
+  sim_secondary_samples <- samples[variable == "sim_secondary"]
+
+  # Calculate summary measures
+  predictions <- calc_summary_measures(
+    sim_secondary_samples,
+    summarise_by = "date",
+    order_by = "date",
+    CrIs = CrIs
+  )
+
+  # Round predictions
+  predictions <- predictions[, lapply(.SD, round, 1)]
```

Comment:
Nothing in the doc string says this will happen. It might be nice to do so but also to allow this to be set by the user.